pkg/apis/maistra/v2/smcp_new.yaml

```diff
@@ -0,0 +1,410 @@
+        ports:
+        - name: status-port
+          port: 15020
+        - name: http2
+          port: 80
+          targetPort: 8080
+        - name: https
+          port: 443
+          targetPort: 8443
+      meshExpansionPorts: [] # additional expansion ports. default expansion ports will be added automatically if multiCluster is configured
+    egress: # _the_ istio-egressgateway
+      service:
+        type: ClusterIP
+        ports:
+        - name: status-port
+          port: 15020
+        - name: http2
+          port: 80
+          targetPort: 8080
+        - name: https
+          port: 443
+          targetPort: 8443
+      # requestedNetworkView: [external] set automatically if multi-cluster is enabled
+    additional:
+      some-gateway: {}
+
+  runtime:
+    # component specific overrides
+    pilot: # same as gateway.runtime structure illustrated above
+      deployment:
+        replicas: 2
+      pod:
+        affinity: {}
+        containers:
+          discovery:
+            resources:
+              limits: {}
+              requirements: {}
+    citadel: {} # similar to pilot above
+    galley: {} # similar to pilot above
+    # note, mixer runtime is configured as part of policy.mixer or telemetry.mixer
+    # default runtime settings
+    defaults: # merged into any component specific overrides. component specific overrides take precedence
+      deployment: {} # similar to gateway.runtime.deployment
+      pod:
+        nodeSelector: {}
+        tolerations: {}
+        priorityClassName: ""
+      container:
+        imageRegistry: registry.redhat.io # old global.hub
+        imageTag: "2.0" # old global.tag
+        imagePullPolicy: Always
+        imagePullSecrets:
+        - name: registry-token-secret
+        resources: # old global.defaultResources
+          limits: {}
+          requirements: {}
+
+  addons:
+    metrics:
+      prometheus:
+        address: some-url-to-prometheus # specify this or install (i'm not sure if we can support this or not)
+        install: # install prometheus, as opposed to using an existing install at address
+          selfManaged: true # use prometheus CR, maybe poor naming?
+          config:
+            retention: 6h
+            scrapeInterval: 15s
+          useTLS: true # .Values.prometheus.security.enabled, i don't think this applies to us
+          service:
+            metadata: {}
+            nodePort: null # use node port if not null
+            ingress: # enables Ingress/Route if not null
+              contextPath: /prometheus
+              tls: # raw yaml, so it can be used with both Route and Ingress resources
+                termination: reencrypt
+          runtime:
+            deployment: {}
+            pod: {}
+
+    tracing:
+      type: Jaeger # or Zipkin, Lightstep, Datadog, Stackdriver
+      jaeger:
+        name: jaeger # name of Jaeger CR to reference/create
+        install: # create a jaeger CR if it doesn't already exist
```

Comment:
how do you trigger installation? just by settings any values under `install`?

Reply:
Yes, if `install` is not null, maistra will create a CR with the settings.  If a CR already exists, no action will be taken.